video2dataset/downloader.py

```diff
@@ -24,19 +25,31 @@ def handle_youtube(youtube_url):
     return cv2_vid, dst_name
 
 
-def handle_mp4_link(mp4_link):
-    resp = requests.get(mp4_link, stream=True)
-    ntf = tempfile.NamedTemporaryFile()  # pylint: disable=consider-using-with
-    ntf.write(resp.content)
-    ntf.seek(0)
-    dst_name = mp4_link.split("/")[-1][:-4] + ".npy"
-    return ntf, dst_name
+def handle_mp4_link(
+    mp4_link: str,
+    extract_audio: bool = False,
+    sample_rate: int = None
+):
+    if extract_audio:
```

Comment:
you should utilize the video bytes for this and also give the option to return both as we discussed in a dict with the extensions as the keys

Reply:
Already implemented, testing now